app/controllers/advertisements_controller.rb

```diff
@@ -0,0 +1,28 @@
+class AdvertisementsController < ApplicationController
+  def index
+    @advertisement = Advertisement.all
+  end
+
+  def show
+    @advertisement = Advertisement.find(params[:id])
+  end
+
+  def new
+    @advertisement = Advertisement.new
+  end
+
+  def create
+    @advertisement = Advertisement.new
```

Comment:
I'm guessing there's no harm is leaving it as an instance variable. As I've done this more and more, I tend to scope things as small as possible to keep them very isolated.